### PR TITLE
Action log

### PR DESCRIPTION
**Pull request by tom-MO**
[Jira Task](https://myoxygenltd.atlassian.net/browse/MYOX-7?atlOrigin=eyJpIjoiZTBhMDJhMjRjNWUyNGE2NzgwZjZkNGYzODEwZTViZWEiLCJwIjoiaiJ9)

This PR looks at implementing `ActionLog` (previously partially implemented in ScholarPack) and an associate log file viewer to allow test users to navigate through the logs. It gives a good indication of what's going on in the app, and allows testers to see what is coming and going from the servers.

The actual widgets allow the user to copy and delete log files as they wish. These files can also be deleted from the device's app settings menu (by doing "clear data/cache"), but visually seening this list of files is clearer to understand, and they are not deleted when they are copied (unlike the ScholarPack implementation).

There is an example project that demonstrates its behaviour and the intended implementation.

### CODE REVIEW

File: action_log/.idea/workspace.xml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (MO-James) [inline action_log/.idea/workspace.xml]: A lot of these files are IntelliJ/AS specific. 
Can you add them to .gitignore if appropriate. 
This one should be ignored https://stackoverflow.com/questions/27647889/should-i-add-idea-workspace-xml-to-gitignore

File: action_log/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+## 0.0.1
+
+* TODO: Describe initial release.
```
Comment (MO-James) [inline action_log/CHANGELOG.md]: Leftover (autogenerated?) todo

File: action_log/LICENSE [hunk 3]
```diff
@@ -0,0 +1 @@
+TODO: Add your license here.
```
Comment (MO-James) [inline action_log/LICENSE]: Leftover (autogenerated?) todo

File: action_log/README.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+```yaml
+action_log:
+  git:
+    url: https://github.com/MyOxygen/MyOxygen-Flutter-Libraries.git
+    path: action_log
+    ref: ActionLog-v0.0.1 # Use the latest ActionLog tag!!
+```
+
+### SnackBarPresenter's APIs
```
Comment (MO-James) [inline action_log/README.md]: This is `ActionLog` not `SnackbarPresenter`

**Review by MO-James — COMMENTED** [review]
Code looks fine for a quick look over. 
Can you remove the `.gitignore`able files? It's hard to do a code review on 84 files when most aren't important.

**Comment by tom-MO** [comment]
@MO-James Addressed your comments. I've removed the `.idea` directory, but that still leaves about 80 files. Most of these are native project files, which are required to build and run. Were there any other specific files/folders you noticed?

**Review by MO-James — APPROVED** [review]
Cool, LGTM

**Pull request merged**